flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -993,9 +1000,34 @@ object JobManager {
       configuration.setString(ConfigConstants.FLINK_BASE_DIR_PATH_KEY, configDir + "/..")
     }
 
-    val hostname = configuration.getString(ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY, null)
-    val port = configuration.getInteger(ConfigConstants.JOB_MANAGER_IPC_PORT_KEY,
-      ConfigConstants.DEFAULT_JOB_MANAGER_IPC_PORT)
+    // HA mode
+    val (hostname, port) = if (ZooKeeperUtil.isJobManagerHighAvailabilityEnabled(configuration)) {
+      // TODO @removeme @tillrohrmann This is the place where the host and random port for JM is
+      // chosen.  For the FlinkMiniCluster you have to choose it on your own.
+      LOG.info("HA mode.")
```

Comment:
How about we make this a bit more prominent ;-) "Staring JobManager in High Availability Mode".

The other case should also have a log message, for example "Starting JobManager without High Availability".